hbase_sql/src/main/java/com/splicemachine/compactions/SpliceDefaultCompactor.java

```diff
@@ -453,19 +462,19 @@ protected boolean performCompaction(Compactor.FileDetails fd, InternalScanner sc
                 if (closeCheckInterval > 0) {
                     bytesWritten += len;
                     if (bytesWritten > closeCheckInterval) {
-                        bytesWritten = 0;
-//                        if (!store.areWritesEnabled()) {
-//                            progress.cancel();
-//                            return false;
-//                        }
+                        bytesWritten = 0; // reset so check whether cancellation is requested in the next <closeCheckInterval>-bytes mark
+                        if ((isSpark && TaskContext.get().isInterrupted()) || (!isSpark && !store.areWritesEnabled())) {
+                            SpliceLogUtils.debug(LOG, "Compaction cancelled");
+                            progress.cancel();
+                            return false;
+                        }
                     }
                 }
             }
-            // Log the progress of long running compactions every minute if
-            // logging at DEBUG level
+            // Log the progress of long-running compactions every minute if logging at DEBUG level
             if (LOG.isDebugEnabled()) {
                 if ((now - lastMillis) >= 60 * 1000) {
-                    LOG.debug("Compaction progress: " + progress + String.format(", rate=%.2f kB/sec",
+                    LOG.debug(String.format("Compaction progress: %s, rate=%.2f kB/sec", progress.toString(),
```

Comment:
Call to toString is redundant, isn't it?

Reply:
fixed.